javalib/src/main/scala/java/util/regex/Matcher.scala

```diff
@@ -275,5 +283,358 @@ object Matcher {
         throw new IllegalStateException("No match available")
       lastMatch
     }
+
+    private[regex] def string(): String = inputstr
+
+    private[regex] def getRegex(): js.RegExp = pattern.newJSRegExp()
+  }
+}
+
+private[regex] trait StartHack {
+  private[regex] def string(): String
+
+  def start(): Int
+
+  private[regex] def getRegex(): js.RegExp
+
+  private[regex] def computeCurrentGroupPositions(): (Int => Int) = {
+    val regex = getRegex
```

Comment:
I also use other properties of regex: so I'm afraid I cannot avoid that.

```
{
      (if (regex.global) "g" else "") +
      (if (regex.ignoreCase) "i" else "") +
      (if (regex.multiline) "m" else "")
    }
```

Reply:
I switched to use the Pattern instead. Good remark. I recover the flags using >> and &, hopefully the flag API should not change.
I wonder however if always setting the flag global to true is the thing to do, since there is no equivalent to global for a pattern. (and Pattern.compile always uses the "g" flag)

Reply:
IIRC, Java's regexes always behave as JS *global* regexes, hence why `Pattern.compile` always uses the `g` flag.